### DISCUSSION TITLE
LittleFS stops appending to files

### DISCUSSION THREAD

**Discussion by romix123** [esp8266/Arduino#8862] [Q&A]
Hi, I have a data logging setup using an ESP-M3 board (but also have test versions on a Wemos D1 mini). 
Every hour, I write some small amount of data to a number of files on LittleFS. 
The data typically is something like:
`['9',  0.49],`
or 
`['11',45433.99],`
After a while, sometimes after much more than 24 hours, but typically at 6:00 in the morning, append to file suddenly stops working. I can't figure out what causes this.
```cpp
void appendFile(const char * path, const char * message) {
File file = LittleFS.open(path, "a");
  if (!file) {…}
if (file.print(message)) {
    debug("Message appended: ");
  } else {
    debugln("Append failed");
  }
  delay(5000);
  file.flush();
  file.close();
}
```
As you can see, I have incorporated a delay, flush and close file. I even unmount and remount the entire set of log updates. Nothing seems to help. Append simply stops working after a while.
I run esp8266 3.1.1, FS is 128K
Any ideas?

**Comment by d-a-v** [comment]
You can try and run your code (and LittleFS library) on your computer and see if this bug is reproducible.
On linux, mac or windows/WSL2 with git installation of this core:
```
$ cd tests/host
$ make help
$ make ../../libraries/LittleFS/examples/SpeedTest/SpeedTest
$ bin32/SpeedTest/SpeedTest
[...]
==> Time to copy 65536 = 65536 bytes = 1 milliseconds = 65.54 MB/s
done
$ make path/to/your/sketch/sketch
```

**Comment by skowron0** [comment]
Have you found a solution to this problem? I have the same problem that after some time I can't add anything to the file. Generally the number of bytes used and free space returns null.

**Comment by romix123** [comment]
NoVerstuurd vanaf mijn iPadOp 10 mrt 2025 om 20:13 heeft skowron0 ***@***.***> het volgende geschreven:﻿
Have you found a solution to this problem? I have the same problem that after some time I can't add anything to the file. Generally the number of bytes used and free space returns null.
